src/cargo/util/config.rs

```diff
@@ -885,23 +895,36 @@ fn walk_tree<F>(pwd: &Path, mut walk: F) -> CargoResult<()>
 }
 
 pub fn save_credentials(cfg: &Config,
-                       token: String) -> CargoResult<()> {
+                        token: String,
+                        registry: Option<String>) -> CargoResult<()> {
     let mut file = {
         cfg.home_path.create_dir()?;
         cfg.home_path.open_rw(Path::new("credentials"), cfg,
-                                   "credentials' config file")?
+                              "credentials' config file")?
+    };
+
+    let (key, value) = {
+        let key = "token".to_string();
+        let value = ConfigValue::String(token, file.path().to_path_buf());
+
+        if let Some(registry) = registry {
+            let mut map = HashMap::new();
+            map.insert(key, value);
+            (registry, CV::Table(map, file.path().to_path_buf()))
```

Comment:
I think this is building up a TOML file like:

```toml
[my-registry-name]
token = "..."
```

right?


That seems fine by me for now (although maybe `registry.foo` for consistency with specifying the index?), but one thing I'm a little worried about is that we're making a lot of decisions in a bit of an ad-hoc fashion. I think all the conclusions are fine so far, but it'd be good to have a list of "here's the design decisions that were made" when we go for stabilizing custom registries.

Mind opening a tracking issue for custom registries (or appending to one that already exists?). I'm thinking we could track things like the `--registry` flag, my concern below for the naming of the registry identifiers, the format of the credentials file, the configuration syntax in `.cargo/config` for custom registries, etc.

Basically I'm thinking we should just track somewhere "here's what we need to be sure about for stabilization" and we can throw concerns/thoughts/implementation details on that as they arise.

Reply:
I will raise an issue to cover this.